src/apis/impl/NotificationApiImpl.ts

```diff
@@ -0,0 +1,79 @@
+import {NotificationApi} from "../NotificationApi";
+import {Notification} from "../../models/Notification";
+import {
+  FilterChainMaintenanceTask,
+  IccHcpartyXApi,
+  IccUserXApi, MaintenanceTask
+} from "@icure/api";
+import {IccMaintenanceTaskXApi} from "@icure/api/icc-x-api/icc-maintenance-task-x-api";
+import {NotificationMapper} from "../../mappers/notification";
+import {PaginatedListNotification} from "../../models/PaginatedListNotification";
+import {Filter} from "../../filter/Filter";
+import {PaginatedListMapper} from "../../mappers/paginatedList";
+import {FilterMapper} from "../../mappers/filter";
+
+export class NotificationApiImpl implements NotificationApi {
+
+  private readonly userApi: IccUserXApi;
+  private readonly maintenanceTaskApi: IccMaintenanceTaskXApi;
+  private readonly hcpApi: IccHcpartyXApi;
+
+  constructor(api: { userApi: IccUserXApi; maintenanceTaskApi: IccMaintenanceTaskXApi, healthcarePartyApi: IccHcpartyXApi}) {
+    this.userApi = api.userApi;
+    this.maintenanceTaskApi = api.maintenanceTaskApi;
+    this.hcpApi = api.healthcarePartyApi;
+  }
+
+  async createOrModifyNotification(notification: Notification, delegate?: string): Promise<Notification | undefined> {
+    return this.userApi.getCurrentUser().then( user => {
+      if(!user) throw new Error("There is no user currently logged in user");
+      const inputMaintenanceTask = NotificationMapper.toMaintenanceTaskDto(notification);
+      if(!inputMaintenanceTask?.rev) {
+        if(!delegate) throw new Error("No delegate provided for Notification creation");
+        return this.maintenanceTaskApi.newInstance(user, inputMaintenanceTask, [delegate])
+          .then( task => {
+            return this.maintenanceTaskApi.createMaintenanceTaskWithUser(user, task).then( (createdTask) => {
+              return NotificationMapper.toNotification(createdTask as MaintenanceTask);
+            });
+          })
+      } else {
+        return this.maintenanceTaskApi.modifyMaintenanceTaskWithUser(user, inputMaintenanceTask).then( (createdTask) => {
```

Comment:
Let's add a check to verify the user is only modifying the status and nothing else